x/ccv/provider/keeper/throttle.go

```diff
@@ -515,29 +541,40 @@ func (k Keeper) GetSlashMeter(ctx sdktypes.Context) sdktypes.Int {
 	store := ctx.KVStore(k.storeKey)
 	bz := store.Get(providertypes.SlashMeterKey())
 	if bz == nil {
+		// Slash meter should be set as a part of InitGenesis and periodically updated by throttle logic,
+		// there is no deletion method exposed, so nil bytes would indicate something is very wrong.
 		panic("slash meter not set")
 	}
 	value := sdktypes.ZeroInt()
 	err := value.Unmarshal(bz)
 	if err != nil {
+		// We should have obtained value bytes that were serialized in SetSlashMeter,
+		// so an error here would indicate something is very wrong.
 		panic(fmt.Sprintf("failed to unmarshal slash meter: %v", err))
 	}
 	return value
 }
 
 // SetSlashMeter sets the slash meter to the given signed int value
 //
-// Note: the value of this int should always be in the range of tendermint's [-MaxVotingPower, MaxVotingPower]
+// Note: the value of this int should always be in the range of tendermint's [-MaxTotalVotingPower, MaxTotalVotingPower]
 func (k Keeper) SetSlashMeter(ctx sdktypes.Context, value sdktypes.Int) {
+	// The following panics are included since they are invariants for slash meter value.
```

Comment:
Could we just set the meter to `MaxTotalVotingPower` in this case?

Reply:
If the slash meter is attempted to be set out of the range of `[-MaxTotalVotingPower, MaxTotalVotingPower]`, either something is incorrect/unexpected with the throttling logic, or someone has found a vulnerability. Allowing binaries to safely carry along in this case is pretty dangerous. 

Regardless, if the throttling logic is correct (and esp our understanding of its properties), then the slash meter should stay in the expected range. We could potentially remove this invariant logic altogether if we're confident in throttling correctness. 

Personally I'm not 100% confident in its correctness until we spec it out and such :)

Reply:
For now, leaving the panics and adding a todo to remove them once https://github.com/cosmos/interchain-security/issues/534 is solved